RiotNSE/NotificationService.swift

```diff
@@ -0,0 +1,499 @@
+            NSLog("[NotificationService] Fallback case 1")
+            contentHandler?(content)
+            return
+        }
+        
+        userAccount.mxSession.event(withEventId: eventId, inRoom: roomId, success: { [weak self] (event) in
+            guard let self = self else {
+                NSLog("[NotificationService] Fallback case 9")
+                return
+            }
+            
+            guard let event = event else {
+                return
+            }
+            
+            guard event.isEncrypted else {
+                //  not encrypted, go on processing
+                self.processEvent(event)
+                return
+            }
+            
+            //  encrypted
+            guard self.showDecryptedContentInNotifications else {
+                //  do not show decrypted content in notification
+                self.fallbackToOriginalContent()
+                return
+            }
+            
+            //  should show decrypted content in notification
+            guard event.clear == nil else {
+                //  already decrypted
+                self.processEvent(event)
+                return
+            }
+            
+            //  should decrypt it first
+            if userAccount.mxSession.decryptEvent(event, inTimeline: nil) {
+                //  decryption succeeded
+                self.processEvent(event)
+            } else {
+                //  decryption failed
+                NSLog("[NotificationService] Event needs to be decrpyted, but we don't have the keys to decrypt it. Launching a background sync.")
+                self.launchBackgroundSync()
+            }
+        }) { [weak self] (error) in
+            guard let self = self else {
+                NSLog("[NotificationService] Fallback case 10")
+                return
+            }
+            NSLog("[NotificationService] Fallback case 3")
+            self.fallbackToOriginalContent()
+        }
+    }
+    
+    func launchBackgroundSync() {
+        guard let userAccount = userAccount else { return }
+        guard let store = store else { return }
+        if userAccount.mxSession == nil {
+            userAccount.openSession(with: store)
+        }
+
+        //  launch an initial background sync
+        userAccount.initialBackgroundSync(20000, success: { [weak self] in
+            guard let self = self else {
+                NSLog("[NotificationService] Fallback case 12")
+                return
+            }
+            self.fetchEvent()
+        }) { [weak self] (error) in
+            guard let self = self else {
+                NSLog("[NotificationService] Fallback case 11")
+                return
+            }
+            NSLog("[NotificationService] Fallback case 6")
+            self.fallbackToOriginalContent()
+        }
+    }
+    
+    func processEvent(_ event: MXEvent) {
+        guard let content = originalContent, let userAccount = userAccount else {
+            return
```

Comment:
Should we call self.fallbackToOriginalContent() here?

Reply:
Adding